src/jdk.compiler/share/classes/com/sun/tools/javac/tree/TreeMaker.java

```diff
@@ -1064,19 +1064,20 @@ public JCVariableDecl Param(Name name, Type argtype, Symbol owner) {
         return VarDef(new VarSymbol(PARAMETER, name, argtype, owner), null);
     }
 
-    /** Create a list of value parameter trees x0, ..., xn from a list of
-     *  their types and their owner.
+    /** Create a list of value parameter trees for a method's parameters.
+     *  If the method has parameter names defined, use those same symbols,
+     *  otherwse create placeholders x0, x1, ..., xn.
      */
-    public List<JCVariableDecl> Params(List<Type> argtypes, Symbol owner) {
+    public List<JCVariableDecl> Params(MethodSymbol mth, List<Type> argtypes) {
```

Comment:
why changing the order of the arguments in the method declaration?

Reply:
The thought was this: we are adding new preconditions to an existing method. All uses of this method in the JDK itself have been verified to already satisfy these preconditions, but there could (in theory) be other uses outside of the JDK that don't. By changing the method signature, any such uses will no longer compile, instead of silently becoming invalid.

Just trying to be conservative, in other words. But maybe this is too hypothetical, in which case I'm happy to switch it back.